### PR TITLE
NAS-113041 / 13.0 / FTP TimeoutNoTransfer

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by themylogin** [comment]
time 01:00

File: src/middlewared/middlewared/alembic/versions/13.0/2021-12-10_13-36_ftp_timeout_notransfer.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table('services_ftp', schema=None) as batch_op:
+        batch_op.add_column(sa.Column('ftp_timeout_notransfer', sa.Integer(), nullable=True))
+
+    op.execute("UPDATE services_ftp SET ftp_timeout_notransfer = 300")
```
Comment (Qubad786) [inline src/middlewared/middlewared/alembic/versions/13.0/2021-12-10_13-36_ftp_timeout_notransfer.py:24]: 💯

File: src/middlewared/middlewared/plugins/ftp.py [hunk 2]
```diff
@@ -71,6 +72,7 @@ async def ftp_extend(self, data):
         Int('ipconnections', validators=[Range(min=0, max=1000)]),
         Int('loginattempt', validators=[Range(min=0, max=1000)]),
         Int('timeout', validators=[Range(min=0, max=10000)]),
+        Int('timeout_notransfer', validators=[Range(min=0, max=10000)]),
```
Comment (Qubad786) [inline src/middlewared/middlewared/plugins/ftp.py:75]: Should we have `default=300` here too?

**Review by Qubad786 — APPROVED** [review]
Just a question otherwise LGTM! 👍

File: src/middlewared/middlewared/alembic/versions/13.0/2021-12-10_13-36_ftp_timeout_notransfer.py (hunk 1 above)
Comment (themylogin) [reply src/middlewared/middlewared/alembic/versions/13.0/2021-12-10_13-36_ftp_timeout_notransfer.py:24]: ?

File: src/middlewared/middlewared/plugins/ftp.py (hunk 2 above)
Comment (themylogin) [reply src/middlewared/middlewared/plugins/ftp.py:75]: What for?

Comment (Qubad786) [reply src/middlewared/middlewared/plugins/ftp.py:75]: just a question as UI might want to skip this field in payload not get validation error 🤔  Not something that I am suggesting – we're good!

File: src/middlewared/middlewared/alembic/versions/13.0/2021-12-10_13-36_ftp_timeout_notransfer.py (hunk 1 above)
Comment (Qubad786) [reply src/middlewared/middlewared/alembic/versions/13.0/2021-12-10_13-36_ftp_timeout_notransfer.py:24]: just a token of appreciation 😄

File: src/middlewared/middlewared/plugins/ftp.py (hunk 2 above)
Comment (themylogin) [reply src/middlewared/middlewared/plugins/ftp.py:75]: @Qubad786 none of the fields are required for `update` methods, they work like HTTP PATCH.

Comment (Qubad786) [reply src/middlewared/middlewared/plugins/ftp.py:75]: Ah makes sense.

**Comment by rick-mesta** [comment]
@themylogin : is this bad boy ready to merge ?

**Comment by themylogin** [comment]
@rick-mesta yes

**Pull request merged**